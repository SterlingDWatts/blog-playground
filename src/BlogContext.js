import React from "react";

const BlogContext = React.createContext({
  blogs: [],
  searchTerm: "",
  sortBy: "",
  handleSubmitSearch: () => {}
});

export default BlogContext;
