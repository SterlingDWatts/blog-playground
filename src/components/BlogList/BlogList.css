.BlogList__item {
  padding: 8px;
  border-bottom: 1px solid black;
}
