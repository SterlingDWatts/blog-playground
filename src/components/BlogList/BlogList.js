import React, { Component } from "react";
import FeaturedBlog from "../FeaturedBlog/FeaturedBlog";
import BlogListItem from "../BlogListItem/BlogListItem";
import BlogContext from "../../BlogContext";
import "./BlogList.css";

class BlogList extends Component {
  static contextType = BlogContext;

  render() {
    const blogs = this.context.blogs;
    const featBlog = blogs.length > 0 ? <FeaturedBlog blog={blogs[0]} /> : " ";
    const blogsList =
      blogs.length > 1
        ? blogs.slice(1).map(blog => <BlogListItem blog={blog} key={blog.id} />)
        : " ";
    return (
      <ul className="BlogList">
        {featBlog}
        {blogsList}
      </ul>
    );
  }
}

export default BlogList;
