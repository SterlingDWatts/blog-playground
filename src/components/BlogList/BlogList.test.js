import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import BlogContext from "../../BlogContext";
import blogs from "../../blog-file";
import BlogList from "./BlogList";

describe(`BlogList Component`, () => {
  // TODO add snapshot with context content

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BlogList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<BlogList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders list items when context is passed in", () => {
    const contextValue = {
      blogs
    };
    const tree = renderer
      .create(
        <BlogContext.Provider value={contextValue}>
          <BlogList />
        </BlogContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
