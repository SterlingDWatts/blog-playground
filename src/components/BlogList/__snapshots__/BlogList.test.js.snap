// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList Component renders list items when context is passed in 1`] = `
<ul
  className="BlogList"
>
   
   
</ul>
`;

exports[`BlogList Component renders the UI as expected 1`] = `
<ul
  className="BlogList"
>
   
   
</ul>
`;
